src/astro/files/locations/gcs.py

```diff
@@ -0,0 +1,34 @@
+from typing import Dict, List
+from urllib.parse import urlparse, urlunparse
+
+from astro.constants import FileLocation
+from astro.files.locations.base import Location
+from astro.utils.dependencies import GCSClient, GCSHook, gcs
+
+
+class GS(Location):
+    """Handler GS object store operations"""
+
+    location_type = FileLocation.GS
+
+    def get_transport_params(self) -> Dict:
+        """get GCS credentials for storage"""
+        if self.conn_id:
+            gcs_hook = GCSHook(self.conn_id)
+            client = gcs_hook.get_conn()  # type: ignore
+        else:
+            client = GCSClient()
```

Comment:
Do we need to build a client? Should we not fallback to default connection id?

```python
    hook = GCSHook(self.conn_id) if self.conn_id else GCSHook()
    client = gcs_hook.get_conn()
```

https://github.com/apache/airflow/blob/27ded333a9cb5452541bd4bd362f18c03b29cfeb/airflow/providers/google/cloud/hooks/gcs.py#L134

Reply:
I'm not sure if I understand this correctly. 

But I think we need to create a client and give users an option to create it with explicit conn_id if they want to.

Reply:
Please let me know If I'm missing something.